gui/widget.cpp

```diff
@@ -201,7 +201,7 @@ void Widget::setEnabled(bool e) {
 		else
 			clearFlags(WIDGET_ENABLED);
 
-		g_gui.scheduleTopDialogRedraw();
```

Comment:
If you change here for all widgets, then all widgets must be adapted not only buttons.
I suspect there is even a bug currently with dropdowns because disabled state draws in foreground without depending on idle state drawing in background.
The way I understand it, if one state of the widget draws on background, every state drawing in foreground should ultimately depend on a state drawing in background.

Reply:
Not every foreground state should depend on a background one.
BUT if you have one state (or more) drawing on background, all states drawing on foreground should depend on a state drawing on background.
Else, I bet we can get troubles in drawing because the foreground would not be rendered the same when the state of the widget change (as some background may not be cleared/rerendered when drawing on foreground).
But yes, having the code spread with manual redraws makes errors quite invisible.